### PR TITLE
Feature/typing and doc

### PR DESCRIPTION
**Pull request by berkbaski**

### CODE REVIEW

**Comment by berkbaski** [comment]
hey @furkanarabaci! I think there is have an access problem because I couldn't add you as a reviewer. Could you add yourself to reviewer?

**Comment by berkbaski** [comment]
Hey @furkanarabaci!

BTW, I updated sample section of the README file. But I'm not sure this is right. The old version has page code. I thought I should update it as a new page.

We can also do it like this maybe.

```typescript
import Image from "sf-core/ui/image";
import extend from "js-base/core/extend";
import Color from 'sf-core/ui/color';
import Label from 'sf-core/ui/label';
import FlexLayout from 'sf-core/ui/flexlayout';
import ListView from 'sf-core/ui/listview';
import ListViewItem from 'sf-core/ui/listviewitem';
import Font from 'sf-core/ui/font';
import TextAlignment from 'sf-core/ui/textalignment';
import ListviewIndex from 'sf-extension-listviewindex';

// ...

const _headerData = ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "R", "S", "T", "U", "V", "W", "X", "Y", "Z", Image.createFromFile("images://trash.png")];

const _rowData = [];
for (let i = 0; i < _headerData.length; i++) {
    _rowData.push(["#ff6c8f", "#ff85a2", "#ff9fb6", "#ffb8c9", "#ffd2dc", "#ffebf0"]);
}

const dataArray = [];

const headerIndex = [];

function pushDataToArray(headerData, rowData) {
    for (let i = 0; i < headerData.length; i++) {
        dataArray.push({ isHeader: true, data: headerData[i] });
        headerIndex.push(dataArray.length - 1);
        for (let j = 0; j < rowData[i].length; j++) {
            dataArray.push({ isHeader: false, data: rowData[i][j] });
        }
    }
};
pushDataToArray(_headerData, _rowData);

const myListView = new ListView({
    flexGrow: 1,
    marginLeft: 20,
    itemCount: dataArray.length,
});

this.layout.addChild(myListView);

const listviewindex = new ListviewIndex();
listviewindex.width = 20;

listviewindex.indexItems = function () {
    return _headerData;
}

listviewindex.indexDidSelect = function (index) {
    myListView.scrollTo(headerIndex[index], false);
    return true; //haptic
}

listviewindex.reloadData();

this.layout.addChild(listviewindex);


// listviewindex.backgroundView.backgroundColor = Color.GREEN;
// listviewindex.backgroundColor = Color.BLUE;
// listviewindex.tintColor = Color.RED;
// listviewindex.itemSpacing = 5;
// listviewindex.font = Font.create("TimesNewRomanPS-BoldMT", 16);
// listviewindex.indexInset = { top: 10, left: 0, bottom: 10, right: 0 };
// listviewindex.indexOffset = { vertical: -2, horizontal: -2 };
// listviewindex.listviewIndexMinimumWidth = 40;

myListView.onRowCreate = function (type) {
    const myListViewItem = new ListViewItem();

    if (type == 1) {
        const myLabelTitle = new Label({
            flexGrow: 1,
            margin: 10
        });
        myLabelTitle.textColor = Color.WHITE;
        myLabelTitle.borderRadius = 10;
        myLabelTitle.textAlignment = TextAlignment.MIDCENTER;
        myListViewItem.addChild(myLabelTitle);
        myListViewItem.myLabelTitle = myLabelTitle;
    }
    else { // Header
        const myLabelTitle = new Label({
            flexGrow: 1,
            margin: 10
        });
        myLabelTitle.font = Font.create(Font.DEFAULT, 30, Font.BOLD);
        myLabelTitle.backgroundColor = Color.WHITE;
        myListViewItem.addChild(myLabelTitle);
        myListViewItem.myLabelTitle = myLabelTitle;
    }

    return myListViewItem;
};

myListView.onRowHeight = function (index) {
    if (dataArray[index].isHeader) {
        return 100;
    }
    return 70;
};

myListView.onRowBind = function (listViewItem, index) {
    const myLabelTitle = listViewItem.myLabelTitle;

    if (dataArray[index].isHeader) {
        myLabelTitle.text = (typeof dataArray[index].data === 'string') ? dataArray[index].data : "Image";
    }
    else {
        myLabelTitle.backgroundColor = Color.create(dataArray[index].data);
        myLabelTitle.text = dataArray[index].data;
    }
};

myListView.onRowType = function (index) {
    if (dataArray[index].isHeader) {
        return 2;
    }
    else {
        return 1;
    }
};

this.layout.flexDirection = FlexLayout.FlexDirection.ROW;
```

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+ * @type {object}
+ * @author Berk Baski <berk.baski@smartface.io>
+ * @copyright Smartface 2021
+ */
+
+import View from "sf-core/ui/view";
+import Color from "sf-core/ui/color";
+import Font from "sf-core/ui/font";
+
+export type IndexInsetType = {
```
Comment (berkbaski) [inline index.d.ts:12]: I'm not sure about it. Maybe we should add this to `sf-core`?

Also, I couldn't add a description here. Do you have a suggestion? @furkanarabaci

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    top: number;
+    left: number;
+    bottom: number;
+    right: number;
+}
+
+export type IndexOffsetType = {
```
Comment (berkbaski) [inline index.d.ts:19]: I'm not sure about it. Maybe we should add this to `sf-core`?

Also, I couldn't add a description here. Do you have a suggestion?

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    indexItems(): void;
+
+    /**
+     * Called as a result of recognizing an index touch. Return true to produce a haptic feedback (iPhone 7 with iOS 10 or later).
+     * @event
+     * @return [boolean]
+     * @since 1.0
+     * @default
+     */
+    indexDidSelect(): void;
```
Comment (furkanarabaci) [inline index.d.ts:126]: This should return boolean

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    listviewIndexMinimumWidth: number;
+
+    /**
+     * Provides a set of items to display in the list index.The library support string images items.
+     * @event
+     * @return [string|UI.Image]
+     * @since 1.0
+     * @default
+     */
+    indexItems(): void;
```
Comment (furkanarabaci) [inline index.d.ts:117]: This should return string|UI.Image

File: index.d.ts (hunk 2 above)
Comment (furkanarabaci) [reply index.d.ts:19]: Nah

File: index.d.ts (hunk 1 above)
Comment (furkanarabaci) [reply index.d.ts:12]: Nah

**Review by furkanarabaci — APPROVED** [review]

**Pull request merged**